### DISCUSSION TITLE
How to execute useSWR once and then manually stop the update?

### DISCUSSION THREAD

**Discussion by spencerchang** [vercel/swr#2088] [Q&A]
I tried several function such as isPause() and onSuccess() and Conditional Fetching
But after those tries the result is can't show page and component at first.

I expect show the page and component at first and then stop the update when i restart it later.
The implement is like a form to search.

Hope someone can give me some hints
Thanks.

**Comment by AllanOliveiraM** [comment]
Hi!

Did you read the documentation? :)

If you are trying to use the `useSWR` to make the first request and then stop revalidation, but keep the cache, you can disable the automatic revalidation modes available like:

```tsx
import useSWRImmutable from 'swr/immutable'

const App = () => {
    const { data } = useSWRImmutable(key, fetcher)

    return ...
}
```

This approach is the same as manually disabling it like:

```tsx
const { data } = useSWR(key, fetcher, {
  revalidateIfStale: false,
  revalidateOnFocus: false,
  revalidateOnReconnect: false
})
```

See [the docs](https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations) for more details.

If you need to keep data between page refreshes [see the local storage based cache](https://swr.vercel.app/docs/advanced/cache#localstorage-based-persistent-cache).

If you are using Next.js, i see one problem with this method [here](https://github.com/vercel/swr/discussions/2091), unresolved still...

I hope this helps 😄
